### PR TITLE
[FEATURE] adds events and loading states

### PR DESCRIPTION
**Pull request by ryanholte**

### CODE REVIEW

**Comment by makepanic** [comment]
Hi,
is this PR ready to be merged or will you continue working on it?
I'll review it later today.

Thanks for all the work 👍 

Could you maybe also rebase all changes into one commit?
Also i'd like to introduce [Conventional Commits](https://conventionalcommits.org/) into this project. That in turn allows us to easily generate a changelog with [standard-version](https://github.com/conventional-changelog/standard-version).
Could you try to reformat the commit messages into the CC format?

After we merge this we could release a 1.0.0 of this package. I'll add some tests beforehand.

File: addon/components/keyword-complete.js [hunk 1]
```diff
@@ -188,23 +188,52 @@ export default Ember.Component.extend({
...
+      }
+    }).finally(() => {
+      if (this.get('_loadSuggestionsId') === loadSuggestionsId) {
+        this.set('isLoadingSuggestions', false);
+      }
     });
   },
 
-  updateSuggestions: observer('filterQuery', 'currentSourceKey', function () {
+  debouceThis: observer('filterQuery', 'currentSourceKey', function () {
```
Comment (makepanic) [inline addon/components/keyword-complete.js]: typo `debounceThis`

File: tests/dummy/app/templates/demos/emoji.hbs [hunk 2]
```diff
@@ -1,82 +1,87 @@
...
+            extractDataString(item) {
+            return `:${item.val}:`;
+            },
+            loadSuggestions(query) {
+            return RSVP.resolve(ENV.APP.EMOJIS.filter(item => `:${item.val}:`.indexOf(query) !== -1));
+            }
+            }
+            }
+            });{{/code-block}}
+    {{/tabs.tab}}
```
Comment (makepanic) [inline tests/dummy/app/templates/demos/emoji.hbs]: could you revert the indentation here?

**Comment by ryanholte** [comment]
👍  I will make these changes soon. :)

**Comment by ryanholte** [comment]
Sorry for the delay. I finally made changes per code review!

**Comment by makepanic** [comment]
👍 Thanks, I'll squash merge your commits, do some cleanup and cut a new version.

**Pull request merged**